controllers/release/release_adapter.go

```diff
@@ -286,6 +292,26 @@ func (a *Adapter) createReleasePipelineRun(releaseStrategy *v1alpha1.ReleaseStra
 	return pipelineRun, nil
 }
 
+// createSnapshotEnvironmentBinding creates a SnapshotEnvironmentBinding for the Release being processed.
+func (a *Adapter) createSnapshotEnvironmentBinding(releasePlanAdmission *v1alpha1.ReleasePlanAdmission,
+	environment *applicationapiv1alpha1.Environment) (*applicationapiv1alpha1.SnapshotEnvironmentBinding, error) {
+	application, components, snapshot, err := a.getSnapshotEnvironmentResources(releasePlanAdmission)
+	if err != nil {
+		return nil, err
+	}
+
+	binding := gitops.NewSnapshotEnvironmentBinding(components, snapshot, environment)
+	err = ctrl.SetControllerReference(application, binding, a.client.Scheme())
+	if err != nil {
+		return nil, err
+	}
+	err = libhandler.SetOwnerAnnotations(a.release, binding)
```

Comment:
This needs an empty line before and a comment stating this is added to watch for status updates of the binding so they can be tracked in the release.